Source/GSImageMagickImageRep.m

```diff
@@ -129,7 +131,9 @@ + (NSArray*) imageRepsWithData: (NSData *)data allImages: (BOOL)allImages
 
   ExceptionInfo *exception = AcquireExceptionInfo();
   ImageInfo *imageinfo = CloneImageInfo(NULL);
-  Image *images, *image;
+  Image *images;
+  Image *image;
+  char signature[32];
```

Comment:
Why is the signature 32 bytes long when you only use 18 of them?

Reply:
As a safety for future uses, we could try to handle more than one signature.
I was looking and there are other files misbehaving, so one could reuse it.

If you wish I could reduce it to exactly 18, I don't think 14bytes are a big issues though.

Reply:
@fredkiefer to appease both, I defined a local constant. So I declare for now just the 18 bytes needed for TGA, but it is more evident that if you tinker with it, you need to update it. And memset will be correct too. Fine?